src/Compilers/Core/Portable/InternalUtilities/EnumerableExtensions.cs

```diff
@@ -235,6 +235,29 @@ public static IReadOnlyCollection<T> ToCollection<T>(this IEnumerable<T> sequenc
             return source.Cast<T?>().FirstOrDefault(static (v, arg) => arg.predicate(v!.Value, arg.arg), (predicate, arg));
         }
 
+        public static T? LastOrDefault<T, TArg>(this IEnumerable<T> source, Func<T, TArg, bool> predicate, TArg arg)
```

Comment:
In that case I can just do a non-argument LastOrDefault and take the lambda allocation since the framework LINQ one will special case for a bunch of various interfaces. This is on a SyntaxTriviaList so I can also do the .Reverse.FirstOrDefault() if you don't mind that.

Reply:
Firstordefault on a reversed list seems good to me :-)